platform/core.netigso/src/org/netbeans/core/netigso/Netigso.java

```diff
@@ -118,7 +118,7 @@ protected void prepare(Lookup lkp, Collection<? extends Module> preregister) {
         if (framework == null) {
             readBundles();
             
-            Map configMap = new HashMap();
+            Map configMap = new HashMap<>();
```

Comment:
Right you are! The problem is that I could not figure out a way to do generics here on the LHS. If you have a suggestion, I'm completely open to hearing it.

Reply:
lets take a look at the usages:
the map is passed to `frameworkFactory.newFramework` which expects `Map<String, String>`, however one entry which is put into the map using the `"netigso.archive" key` is not a String.

This indicates that this is either a bug or a hack.

https://github.com/search?q=repo%3Aapache%2Fnetbeans%20%22netigso.archive%22&type=code search shows that the key is used and that the type is expected to be in fact not a String.

There is likely no way to set a type to the map without making more changes in other places, since the map is not used in a typesafe manner.

Reply:
I agree. I'm starting to see a lot of things like this, whereby the code clearly needs some work. Thanks for the help.